### PR TITLE
Fix deprecated parameters in `D2VTransformer` and `W2VTransformer`. Fix #1937

### LINKED ISSUE #1937: sklearn_api.d2vmodel.D2VTransformer uses deprecated arguments size and iter
#### Description
The D2VTransformer in `sklearn_api.d2vmodel` uses the deprecated arguments `size` instead of `vector_size` and `iter` instead of `epochs`. According to the warnings this will be breaking in version 4.0.0.

#### Code to Reproduce
```
from gensim.sklearn_api.d2vmodel import D2VTransformer
from gensim.models.doc2vec import TaggedDocument
d2v = D2VTransformer(size=1, iter=1).fit([TaggedDocument(['a','a','a','a','a'], [0])]) # gives warnings
d2v = D2VTransformer(vector_size=1, epochs=1).fit([TaggedDocument(['a','a','a','a','a'], [0])]) #gives errors
```

#### Resulting warnings and errors:
```
/lib/python3.6/site-packages/gensim/models/doc2vec.py:355: UserWarning: The parameter `iter` is deprecated, will be removed in 4.0.0, use `epochs` instead.
  warnings.warn("The parameter `iter` is deprecated, will be removed in 4.0.0, use `epochs` instead.")
/lib/python3.6/site-packages/gensim/models/doc2vec.py:359: UserWarning: The parameter `size` is deprecated, will be removed in 4.0.0, use `vector_size` instead.
  warnings.warn("The parameter `size` is deprecated, will be removed in 4.0.0, use `vector_size` instead.")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-108-561949d569bd> in <module>()
      2 from gensim.models.doc2vec import TaggedDocument
      3 d2v = D2VTransformer(size=1, iter=1).fit([TaggedDocument(['a','a','a','a','a'], [0])])
----> 4 d2v = D2VTransformer(vector_size=1, epochs=1).fit([TaggedDocument(['a','a','a','a','a'], [0])])

TypeError: __init__() got an unexpected keyword argument 'vector_size'
```

#### Versions
```
Linux-4.13.0-36-generic-x86_64-with-Ubuntu-17.10-artful
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0]
NumPy 1.14.1
SciPy 1.0.0
gensim 3.3.0
FAST_VERSION 1
```

### PR DESCRIPTION
**Pull request by MritunjayMohitesh**
I tried modifying a liitle bit of @ swierh code and added a bit what he recommended.

### CODE REVIEW

**Comment by menshikh-iv** [comment]
@MritunjayMohitesh next time please push all code to one PR (no need to create one more for fixes).

**Comment by MritunjayMohitesh** [comment]
Sorry about that.
I will do from the next time. It was actually my first time.

**Comment by menshikh-iv** [comment]
@MritunjayMohitesh I already closed previous PR, let's continue here

File: gensim/models/doc2vec.py [hunk 1]
```diff
@@ -404,7 +400,7 @@ def __init__(self, documents=None, dm_mean=None, dm=1, dbow_words=0, dm_concat=0
             self.train(
                 documents, total_examples=self.corpus_count, epochs=self.epochs,
                 start_alpha=self.alpha, end_alpha=self.min_alpha, callbacks=callbacks)
-
+
```
Comment (menshikh-iv) [inline gensim/models/doc2vec.py]: This file is not relevant to PR, please remove it from PR

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -411,25 +411,21 @@ def score_cbow_pair(model, word, l1):
...
-
     The model can be stored/loaded via its :meth:`~gensim.models.word2vec.Word2Vec.save()` and
     :meth:`~gensim.models.word2vec.Word2Vec.load()` methods, or stored/loaded in a format
     compatible with the original word2vec implementation via `wv.save_word2vec_format()`
     and `Word2VecKeyedVectors.load_word2vec_format()`.
-
     """
 
-    def __init__(self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
+    def __init__(self, sentences=None, size=None, vector_size=100, alpha=0.025, window=5, min_count=5,
```
Comment (menshikh-iv) [inline gensim/models/word2vec.py]: This changes non-relevant too, please remove it.

File: gensim/sklearn_api/w2vmodel.py [hunk 3]
```diff
@@ -23,14 +24,22 @@ class W2VTransformer(TransformerMixin, BaseEstimator):
...
 
-    def __init__(self, size=100, alpha=0.025, window=5, min_count=5, max_vocab_size=None, sample=1e-3, seed=1,
-                 workers=3, min_alpha=0.0001, sg=0, hs=0, negative=5, cbow_mean=1, hashfxn=hash, iter=5, null_word=0,
+    def __init__(self, size=None, vector_size=100, alpha=0.025, window=5, min_count=5, max_vocab_size=None, sample=1e-3, seed=1,
+                 workers=3, min_alpha=0.0001, sg=0, hs=0, negative=5, cbow_mean=1, hashfxn=hash, iter=None, epochs=5, null_word=0,
                  trim_rule=None, sorted_vocab=1, batch_words=10000):
         """
         Sklearn wrapper for Word2Vec model. See gensim.models.Word2Vec for parameter details.
         """
+        if iter is not None:
```
Comment (menshikh-iv) [inline gensim/sklearn_api/w2vmodel.py]: Does it make sense here to rename the arguments here? You can simply pass correct variants to "original" class and this is enough to avoid deprecation problems.

CC: @manneshiva

**Comment by menshikh-iv** [comment]
ping @MritunjayMohitesh

**Comment by MritunjayMohitesh** [comment]
I can't understand why travisci is failing. Some help would be appreciated.

File: gensim/models/doc2vec.py [hunk 4]
```diff
@@ -239,9 +239,9 @@ class TaggedDocument(namedtuple('TaggedDocument', 'words tags')):
     and `tags` (a list of tokens). Tags may be one or more unicode string
     tokens, but typical practice (which will also be most memory-efficient) is
     for the tags list to include a unique integer id as the only tag.
-
+
```
Comment (menshikh-iv) [inline gensim/models/doc2vec.py]: Unrelated change (leading space in empty line), please revert all changes for `doc2vec.py` and Travis will pass successfully

File: gensim/sklearn_api/w2vmodel.py [hunk 5]
```diff
@@ -44,23 +52,23 @@ def __init__(self, size=100, alpha=0.025, window=5, min_count=5, max_vocab_size=
...
         self.cbow_mean = int(cbow_mean)
         self.hashfxn = hashfxn
-        self.iter = iter
+        self.epochs = epochs
         self.null_word = null_word
         self.trim_rule = trim_rule
         self.sorted_vocab = sorted_vocab
         self.batch_words = batch_words
-
+
```
Comment (menshikh-iv) [inline gensim/sklearn_api/w2vmodel.py]: leading spaces again

File: gensim/sklearn_api/w2vmodel.py [hunk 6]
```diff
@@ -44,23 +52,23 @@ def __init__(self, size=100, alpha=0.025, window=5, min_count=5, max_vocab_size=
...
-
+        
     def fit(self, X, y=None):
         """
         Fit the model according to the given training data.
         Calls gensim.models.Word2Vec
         """
         self.gensim_model = models.Word2Vec(
-            sentences=X, size=self.size, alpha=self.alpha,
+            sentences=X, size=self.vector_size, alpha=self.alpha,
```
Comment (menshikh-iv) [inline gensim/sklearn_api/w2vmodel.py]: looks like you should change `size` to `vector_size` and same for `iter` here for avoid DeprecationWarning.

File: gensim/sklearn_api/w2vmodel.py [hunk 7]
```diff
@@ -23,14 +24,21 @@ class W2VTransformer(TransformerMixin, BaseEstimator):
     Base Word2Vec module
     """
 
-    def __init__(self, size=100, alpha=0.025, window=5, min_count=5, max_vocab_size=None, sample=1e-3, seed=1,
-                 workers=3, min_alpha=0.0001, sg=0, hs=0, negative=5, cbow_mean=1, hashfxn=hash, iter=5, null_word=0,
-                 trim_rule=None, sorted_vocab=1, batch_words=10000):
+    def __init__(self, size=None, vector_size=100, alpha=0.025, window=5, min_count=5, max_vocab_size=None, sample=1e-3,
```
Comment (menshikh-iv) [inline gensim/sklearn_api/w2vmodel.py]: No need any changes to `w2vmodel.py` (this deprecation only for d2v), this is also unrelated change, please revert this file.

File: gensim/sklearn_api/d2vmodel.py [hunk 8]
```diff
@@ -24,12 +24,18 @@ class D2VTransformer(TransformerMixin, BaseEstimator):
     """
 
     def __init__(self, dm_mean=None, dm=1, dbow_words=0, dm_concat=0, dm_tag_count=1, docvecs=None,
-                 docvecs_mapfile=None, comment=None, trim_rule=None, size=100, alpha=0.025, window=5, min_count=5,
-                 max_vocab_size=None, sample=1e-3, seed=1, workers=3, min_alpha=0.0001, hs=0, negative=5, cbow_mean=1,
-                 hashfxn=hash, iter=5, sorted_vocab=1, batch_words=10000):
+                 docvecs_mapfile=None, comment=None, trim_rule=None, size=None, vector_size=100, alpha=0.025, window=5,
```
Comment (menshikh-iv) [inline gensim/sklearn_api/d2vmodel.py]: No need to add new arguments/attributes/raise more deprecation stuff, simply change arguments name in `fit` when you pass it to `Doc2Vec`, that's all fix here.

File: gensim/sklearn_api/d2vmodel.py [hunk 9]
```diff
@@ -72,11 +79,11 @@ def fit(self, X, y=None):
             documents=d2v_sentences, dm_mean=self.dm_mean, dm=self.dm,
             dbow_words=self.dbow_words, dm_concat=self.dm_concat, dm_tag_count=self.dm_tag_count,
             docvecs=self.docvecs, docvecs_mapfile=self.docvecs_mapfile, comment=self.comment,
-            trim_rule=self.trim_rule, size=self.size, alpha=self.alpha, window=self.window,
+            trim_rule=self.trim_rule, size=self.vector_size, alpha=self.alpha, window=self.window,
```
Comment (menshikh-iv) [inline gensim/sklearn_api/d2vmodel.py]: All changes of your PR should be in 2 lines of `fit` function, that's all that you needed here.

**Comment by menshikh-iv** [comment]
@MritunjayMohitesh Horaay :fire:, looks like needed change, let's wait CI

**Comment by MritunjayMohitesh** [comment]
first of all, thank you for your help.
Doing this will not give deprecation warning but it should still give the
error as @swierh <https://github.com/swierh> stated since , vector_size(or
epoch) hasn't been passed as argument to W2VTransformer class.
Please correct me if I'm wrong.

**Comment by menshikh-iv** [comment]
@MritunjayMohitesh this shouldn't be passed to w2v class (this have no this parameter). Test passed correctly (probably @swierh trying to run tests based on old version of your PR), right now all works fine.
